tests/protocol/process_order_match.ts

```diff
@@ -33,6 +33,63 @@ describe("Matching Crank", () => {
     );
   });
 
+  it("Failure: queue head is not taker", async () => {
+    // GIVEN
+
+    // Create market, purchaser
+    const [purchaserA, purchaserB, market] = await Promise.all([
+      createWalletWithBalance(monaco.provider),
+      createWalletWithBalance(monaco.provider),
+      monaco.create3WayMarket([3.0]),
+    ]);
+    await market.airdrop(purchaserA, 100.0);
+    await market.airdrop(purchaserB, 100.0);
+
+    const againstPk = await market.againstOrder(1, 10, 3.0, purchaserA);
+    await market.forOrder(1, 10, 3.0, purchaserB);
+
+    await market.processMatchingQueueTakerOnce(againstPk).then(
+      function (_) {
+        assert.fail("This test should have thrown an error");
+      },
+      function (e: AnchorError) {
+        assert.equal(
+          e.logs[e.logs.length - 3],
+          "Program log: AnchorError caused by account: market_matching_queue. Error Code: MatchingQueueHeadNotTaker. Error Number: 6072. Error Message: Core Matching: market matching queue head is not taker.",
+        );
+      },
+    );
+  });
+
+  it("Failure: queue head is not maker", async () => {
+    // GIVEN
+
+    // Create market, purchaser
+    const [purchaserA, purchaserB, market] = await Promise.all([
+      createWalletWithBalance(monaco.provider),
+      createWalletWithBalance(monaco.provider),
+      monaco.create3WayMarket([3.0]),
+    ]);
+    await market.airdrop(purchaserA, 100.0);
+    await market.airdrop(purchaserB, 100.0);
+
+    await market.againstOrder(1, 10, 3.0, purchaserA);
+    await market.forOrder(1, 20, 3.0, purchaserB);
+
+    await market.processMatchingQueueMakerOnce(false, 1, 3.0);
+    await market.processMatchingQueueMakerOnce(true, 1, 3.0).then(
+      function (_) {
+        assert.fail("This test should have thrown an error");
+      },
+      function (e: AnchorError) {
+        assert.equal(
```

Comment:
any reason to not use this? we do it elsewhere
```typescript
function (e: AnchorError) {
    assert.equal(
      e.error.errorCode.code,
      "MatchingQueueHeadNotMaker",
    );
},
```

Reply:
Yeah. Transaction is being executed differently so the error is not coming nicely wrapped as `AnchorError`. I suppose the error here is that I'm casting it to `AnchorError` which it isn't.

Reply:
Specifically this method uses `executeTransactionMaxCompute(instruction)` instead just calling `rpc()` on a method. Not sure what was the decision behind that for testing but I've decided not to change that.